### PR TITLE
log: add klog/v2

### LINKED ISSUE #924: Add klogv2 adapter for logging
Right now we only have a Logrus impl. Add klogv2 since it implements the logr interface.

### PR DESCRIPTION
**Pull request by pires**
Fixes #924

### CODE REVIEW

File: log/klogv2/klogv2.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+//
+// We recommend reading the klog conventions to build a better understanding of levels and when they should apply
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md
+package klogv2
+
+import (
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"k8s.io/klog/v2"
+)
+
```
Comment (sargun) [inline log/klogv2/klogv2.go:28]: What do you think about doing a:

```go
var _ log.Logger = (*adapter)(nil)
```

Comment (sargun) [reply log/klogv2/klogv2.go:28]: (rather than the test in the other file)

**Review by sargun — APPROVED** [review]

File: log/klogv2/klogv2.go (hunk 1 above)
Comment (pires) [reply log/klogv2/klogv2.go:28]: Just following what was there but yeah I'm all in for compile time validation. On it.

**Review by sargun — APPROVED** [review]

**Comment by pires** [comment]
Thank you, @sargun

**Pull request merged**